### PR TITLE
Hc/rum 642/update raygun client user interface

### PR DESCRIPTION
**Pull request by RaygunDaniel**
This branch has changed the way that the user accessible methods are laid out in the react native provider.

This includes providing getters as well as setters for each accessible part of the code and removing methods that should be accessible.

### CODE REVIEW

File: sdk/src/CrashReporter.ts [hunk 1]
```diff
@@ -81,60 +87,50 @@ export default class CrashReporter {
...
 
   /**
-   * Set the crash reporters user object
-   * @param newUser
+   * Removes all breadcrumbs.
    */
-  setUser(newUser: User) {
-    this.user = newUser;
+  clearBreadcrumbs() {
+    this.breadcrumbs = [];
```
Comment (RaygunDaniel) [inline sdk/src/CrashReporter.ts:126]: Needs to send through to bridge

File: sdk/src/CrashReporter.ts [hunk 2]
```diff
@@ -209,12 +205,12 @@ export default class CrashReporter {
...
       ? await symbolicateStackTrace(stackFrames)
-      : { stack: cleanFilePath(stackFrames) };
+      : {stack: cleanFilePath(stackFrames)};
 
     const stack = cleanedStackFrames || [].filter(filterOutReactFrames).map(noAddressAt);
 
     if (isFatal) {
-      this.tags.add('UnhandledError');
+      setCurrentTags(getCurrentTags().concat('UnhandledError'))
     }
```
Comment (RaygunDaniel) [inline sdk/src/CrashReporter.ts:214]: Add "Unhandled error" tag whenever an error enters this method but also append "Fatal" if it is a fatal error.

File: sdk/src/RaygunClient.ts [hunk 3]
```diff
@@ -120,60 +120,37 @@ const init = (raygunClientOptions: RaygunClientOptions) => {
...
  */
-const addTag = (...tags: string[]) => {
-  tags.forEach(tag => {
-    currentTags.add(tag);
-  });
-
-  //Apply tags change to crash reporter
-  if (CrashReportingAvailable("addTags")) crashReporter.addTags(tags);
-
+const setTags = (tags : string[]) => {
```
Comment (RaygunDaniel) [inline sdk/src/RaygunClient.ts:123]: Use ... syntax

File: sdk/src/RaygunClient.ts [hunk 4]
```diff
@@ -227,34 +220,34 @@ const sendError = async (error: Error, ...params: any) => {
...
 /**
- * Apply some transformation lambda to all of the user's custom data.
- * @param updater - The transformation.
+ * Appends custom data to the current set of custom data.
+ * @param customData - The custom data to append
  */
-const updateCustomData = (updater: (customData: CustomData) => CustomData) => {
-  if (!CrashReportingAvailable('updateCustomData')) return;
-  crashReporter.updateCustomData(updater);
+const getCustomData = (customData: CustomData | null) => {
```
Comment (RaygunDaniel) [inline sdk/src/RaygunClient.ts:232]: Shouldnt take arguments

File: sdk/src/Types.ts [hunk 5]
```diff
@@ -30,6 +31,11 @@ export type User = {
   uuid?: string;
 };
 
+export const anonUser: User = {
+  identifier: `anonymous-${getDeviceBasedId()}`,
```
Comment (RaygunDaniel) [inline sdk/src/Types.ts:35]: remove "anonymous-" from this identifier

File: sdk/src/Utils.ts [hunk 6]
```diff
@@ -14,6 +38,7 @@ export const getDeviceBasedId = () =>
     .replace('.', '')}`;
 
 /**
+ * TODO: Remove usage from CrashReporter
```
Comment (RaygunDaniel) [inline sdk/src/Utils.ts:41]: Remove this

**Review by mduncan26 — APPROVED** [review]

**Pull request merged**